molgenis-core-ui/README.md

```diff
@@ -0,0 +1,83 @@
+# MOLGENIS core UI
+
+## Javascript
+We use the [frontend maven plugin](https://github.com/eirslett/frontend-maven-plugin)
+to allow maven to install node and run npm and other javascript tools.
+
+The javascript files in and other static resources src/main/javascript are packed into bundles using webpack.
+
+### Directory structure
+```
+|-- molgenis-core-ui
+    |-- README.md // this file
+    |-- gulpfile.js // javascript build actions, used to write version from pom.xml to package.json
+    |-- package.json // describes what npm packages are used
+    |-- pom.xml // the maven buildfile
+    |-- webpack.config.js // webpack configuration, lists the entry points for the bundles and the loaders
+    |-- webpack.test.config.js // webpack configuration for tape tests
+    |-- node // node gets installed here by frontend maven plugin
+    |   |-- node
+    |   |-- npm
+    |   |-- npm.cmd
+    |-- node_modules // npm modules listed in package.json get downloaded to here by npm and read by webpack and node
+    |   |-- [...]
+    |-- src
+    |   |-- main
+    |   |   |-- [...]
+    |   |   |-- javascript
+    |   |   |   |-- molgenis-global-ui-webpack.js // bundle files specify the contents of the bundles
+    |   |   |   |-- molgenis-global-webpack.js
+    |   |   |   |-- molgenis-vendor-webpack.js
+    |   |   |   |-- modules // contain the source files for the molgenis modules, interpreted by the babel loader, included in the bundles
+    |   |   |   |   |-- i18n
+    |   |   |   |   |   |-- I18nStrings.js
+    |   |   |   |   |-- react-components
+    |   |   |   |   |   |-- AggregateTable.js
+    |   |   |   |   |   |-- AlertMessage.js
+    |   |   |   |   |   |-- [...]
+    |   |   |   |   |   |-- index.js
+    |   |   |   |   |   |-- css // if css files are required by the javascript files in the module, webpack bundles them
```

Comment:
webpacking css is tricky because the 'cascading' bit means that the order in which you include them matters. Imports specify what you need and only care that the import happens before you get  executed. So these two models are at odds. For this reason, we've kept bootstrap.css and the bootstrap theme separate. Require-ing it causes more trouble than it solves.
For small components it's nice to require their css and the style.

We need to make some rules for css, also for what syntax we use for the components. But again, I'd like to keep that out of the story.